### PR TITLE
Fix #19792 copy paste err in EventData

### LINKED ISSUE #19792: Copy-paste error: AfterControllerActionMethodEventData indexes Controller twice, but misses Arguments
### Describe the bug
While reading through the source of MvcDiagnostics.cs I noticed on line 527-530
https://github.com/dotnet/aspnetcore/blob/01c1d0613ec40c6bc12bdbdebfe7b03bd878868a/src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs#L527-530
that 
```
            1 => new KeyValuePair<string, object>(nameof(Controller), Controller),
```
should probably be
```
            1 => new KeyValuePair<string, object>(nameof(Arguments), Arguments),
```
all other EventData derived classes index the constructor parameters, this was the only one that was different (copy paste error?)

### To Reproduce
N/A

### Further technical details
- ASP.NET Core version: 3.0.0+ (according to the github source)

### PR DESCRIPTION
**Pull request by rjvdboon**
Copy-paste error: AfterControllerActionMethodEventData indexes Controller twice, but misses Arguments (#19792)

Addresses #19792

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
@javiercn can you please take a look at this? Thanks!

File: src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs
Comment (pranavkm) [inline src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs:528]: @davidfowl I'm not really familiar with how Diagnostics works. Would this break consumers that indexed in to this?

Comment (javiercn) [reply src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs:528]: I think that the behavior here is wrong, we should fix it, and make an announcement if necessary.

**Review by javiercn — APPROVED** [review]

File: src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs
Comment (davidfowl) [reply src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs:528]: It will break consumers and before we merge it, we should find the big consumers and see how hard it will break

Comment (javiercn) [reply src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs:528]: > It will break consumers and before we merge it, we should find the big consumers and see how hard it will break

How do you propose we do this? Who are the consumers of this data, libraries like App insights and folks like that? 

From what I understand this is diagnostic source, which is very specialized.

Comment (davidfowl) [reply src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs:528]: Application insights, open telemetry, other APM vendors that are consuming this data.

Comment (javiercn) [reply src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs:528]: Good to know.

Given that's the case, maybe we should probably hold-off on this for the time being since we haven't received an actual customer report where this was problematic.

**Comment by mkArtakMSFT** [comment]
@javiercn what are we waiting for here? Is this ready to be merged?

**Comment by javiercn** [comment]
> @javiercn what are we waiting for here? Is this ready to be merged?

https://github.com/dotnet/aspnetcore/pull/19811#discussion_r466521806

File: src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs
Comment (mkArtakMSFT) [reply src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs:528]: We shouldn't keep a PR around if we don't plan to accept that now. If this is ready to go and it does add value (which based on the issue it's referencing seems to be the case) we should take this. Otherwise - let's close the PR.

Comment (javiercn) [reply src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs:528]: Fair enough, let's close then

**Comment by javiercn** [comment]
@rjvdboon thanks for your contribution.

Unfortunately we won't be taking this change at this time since we think there's a small chance it breaks existing telemetry and we want to avoid that at all costs.

Thanks

**Pull request closed without merging**
